### DISCUSSION TITLE
Can't import the 'Battery Healthy' dashboard on Grafana

### DISCUSSION THREAD

**Discussion by alone-dancer** [jheredianet/Teslamate-CustomGrafanaDashboards#26] [Q&A] [answered]
Hello,
This is a great project. But I have a issue.

1. When importing the dashboard by importing jason, the following error occurs: Import failed: JSON -> JS Serialization failed: Unexpected token '<',"<!DOCTYPE"…is not valid JASON
2. Importing the dashboard by pasting the raw code is successful, but all the section of the dashboard displays "No data", and an error is displayed in the upper left corner of each section: db query error:xx:invalid input syntax for type Jason

Anyone can help me to fix it.

**Comment by jheredianet** [comment]
Hi, 
Could you confirm, in order to have context, what Kind of installation of TeslaMate are you using? (docker/standalone). 
It's strange about the option 1, I guess you download the file locally and import as file. Option 2 should be the same, because is the same file. So, if it loads from RAW, it should work, unless you are using an instance of Grafana which data source is not TeslaMate (postgressql database). 
Can you confirm that you are having this issue only with the Battery health dashboard or it's the same behaviour with the other ones?

**Comment by alone-dancer** [comment]
Thanks for your reply.
I installed Teslamate on my NAS using docker-compose. It is installed according to the official instructions. Here is the link
[https://docs.teslamate.org/docs/installation/docker/](url)

As you said， I downloaded the file ‘BatteryHealth.json' locally and imported it as a file for option 1. Is this incorrect?

To clarify, the default dashboards work well. Here are the pictures.
![Teslamate-Overview](https://user-images.githubusercontent.com/117016601/217535807-c0f370b7-8134-49ef-914a-b7a65c7e456f.jpg)
![Battery Health](https://user-images.githubusercontent.com/117016601/217536021-debf6743-75a4-4eee-9c83-f8e5ee324b35.jpg)

**Comment by jheredianet** [comment]
Hello, it's so strange.  The other thing I was wondering is if you have tried to import other dashboards to see if the same thing happens to you. Let me know. 
About the Battery Health, on the screen that you show I see something curious and that is that it has read the current soc at the bottom. But not the rest. Could you go to the setting of the dashboard and see if the variables are getting values? Is the right DataSource selected?

I let you some screenshots of what to check. 
![image](https://user-images.githubusercontent.com/45735191/217539402-81e1da2e-5c18-42c4-80d1-890ec5b97dfa.png)

![image](https://user-images.githubusercontent.com/45735191/217539473-79d18873-059f-41f4-8074-20ac1f11c657.png)

![image](https://user-images.githubusercontent.com/45735191/217539510-a71ba963-e6e5-48f0-9e58-6587340b2ac7.png)

**Reply by alone-dancer** [reply]
Thank you so much. 
I don't know how to enter the Battery Health page.
![Battery Health-setting](https://user-images.githubusercontent.com/117016601/217548514-f34d7a91-7425-46cf-b63b-705b86b607a3.jpg)

I'll study it and reply to you.

**Reply by alone-dancer** [reply]
Thank you so much. 
I don't know how to enter the Battery Health page.
![Battery Health-setting](https://user-images.githubusercontent.com/117016601/217548514-f34d7a91-7425-46cf-b63b-705b86b607a3.jpg)

I'll study it and reply to you.

**Reply by jheredianet** [reply]
Here you have it:

![image](https://user-images.githubusercontent.com/45735191/217562766-88e9f74a-f5f5-4af2-a5cf-c70de80b1207.png)

**Comment by alone-dancer** [comment]
I imported the 'Charging Costs Stats' and 'ChargingCurveStats'. Seems they work well.
![Charging Costs Stats](https://user-images.githubusercontent.com/117016601/217544710-57e18b04-fd01-499f-8d02-f322e31555e9.jpg)

**Reply by jheredianet** [reply]
Oh!!! wait, By the way. Is your car new? or just begun using Teslamate?, as I can see you only have one charge. Maybe that's why you can't see any data on the battery health dashboard, because all the estimated values come from the charge sessions you've made, as explained here https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards#battery-health

**Reply by alone-dancer** [reply]
I see. 
I just used Teslamate last week.During this period, it was only charged once. I think that's the reason for the issue.
It should be OK to use it for a few more days。
Really appreciate your help.

**Reply by jheredianet** [reply]
You're welcome. Just continue using your car and Teslamate a few more days. Cheers!

**Comment by jheredianet** [comment, marked as answer]
In the other hand, have you tried to Auto import these dashboards as indicated here?
https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards#how-to-auto-import-these-custom-dashboards

Is other way to have all the dashboards together and updated too.

**Reply by alone-dancer** [reply]
I will check whether the board 'Battery Health' work normally after a few more days, , and then I will try to import it automatically.
Thank you again.

**Comment by jayali786** [comment]
My car is used and i am using this past 10 days, I am getting same issue error,  and i also check the value and it showing nothing there

**Reply by alone-dancer** [reply]
10 days may be too short. My Teslamate has been running for more than 30 days (charging once a week), and still displays No data.

**Reply by jheredianet** [reply]
> My car is used and i am using this past 10 days, I am getting same issue error, and i also check the value and it showing nothing there

Hi, as it is in the panel description, it needs a few drives and especially some charging sessions (of more than 15kWh) to start doing the calculations. Also keep in mind that the possible degradation will only be a reference since you started using Teslamate, it is not the total degradation of the battery car's life time.

**Comment by jheredianet** [comment]
Hi guys, in case you are interested, I've released a new version of this dashboard...
There is an important improvement for those who have not used Teslamate since they got their car, or for those who have bought it second hand. Now it is possible to set the max range to 100% and the battery capacity of the car battery when it was new in order to get an accurate and better estimation.

**Comment by alone-dancer** [comment]
That's great. It works well.
Thanks so much for the update.

**Reply by jheredianet** [reply]
Glad to hear that. Cheers!!!
